assembly/src/assembler/context.rs

```diff
@@ -77,6 +93,16 @@ impl AssemblyContext {
         self.current_proc_context().expect("no procedures").num_locals
     }
 
+    /// Returns the reference to the procedures map.
+    pub fn proc_map(&self) -> &BTreeMap<ProcedureId, ProcedureName> {
+        &self.proc_map
+    }
```

Comment:
Instead of exposing procedure map, I would provide a dedicated function for getting procedure names. For example:
```Rust
pub fn get_imported_procedure_name(&self, &ProcedureId) -> Option<&ProcedureName>
```

This function then could get this info from the `ModuleContext` which is currently on the top of the stack.